src/interaction/FixedLocalTupleComListInteractionTemplate.hpp

```diff
@@ -0,0 +1,408 @@
+/*
+  Copyright (C) 2017
+      Max Planck Institute for Polymer Research
+  
+  This file is part of ESPResSo++.
+  
+  ESPResSo++ is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+
+// ESPP_CLASS
+#ifndef _INTERACTION_FIXEDLOCALTUPLECOMLISTINTERACTIONTEMPLATE_HPP
+#define _INTERACTION_FIXEDLOCALTUPLECOMLISTINTERACTIONTEMPLATE_HPP
+
+#include "mpi.hpp"
+#include "Interaction.hpp"
+#include "Real3D.hpp"
+#include "Tensor.hpp"
+#include "Particle.hpp"
+#include "FixedLocalTupleList.hpp"
+#include "esutil/Array2D.hpp"
+#include "bc/BC.hpp"
+#include "SystemAccess.hpp"
+#include "storage/Storage.hpp"
+#include "Interaction.hpp"
+#include "types.hpp"
+#include "esutil/Error.hpp"
+
+namespace espressopp {
+    namespace interaction {
+	template < typename _Potential >
+	class FixedLocalTupleComListInteractionTemplate: public Interaction, SystemAccess {
+	    
+	protected:
+	    typedef _Potential Potential;
+	    
+	private:
+	    long unsigned int num_of_part; // total number of particles
+	    long unsigned int num_of_subchain; // total number of subchains
+	    std::map<long unsigned int, Real3D> com_origin;
+	    
+	    int N_Constrain; // number of particles constraining the interaction
+	    int nParticles;  // number of particles in this node
+	    
+	public:
+	    FixedLocalTupleComListInteractionTemplate
+	    (shared_ptr < System > _system,
+	     shared_ptr < FixedLocalTupleList > _fixedtupleList,
+	     shared_ptr < Potential > _potential)
+	      : SystemAccess(_system), fixedtupleList(_fixedtupleList), potential(_potential) {
+		if (! potential) {
+		    LOG4ESPP_ERROR(theLogger, "NULL potential");
+		}
+		System& system = getSystemRef();
+		esutil::Error err(system.comm);
+
+		long unsigned int localN = system.storage -> getNRealParticles();
+		boost::mpi::all_reduce(*system.comm, localN, num_of_part, std::plus<int>());
+
+		FixedLocalTupleList::TupleList::Iterator it(*fixedtupleList);
+		std::vector<Particle*> pList = it->second;
+
+		N_Constrain = pList.size() + 1;
```

Comment:
According with specifications of `FixedLocalTupleList`, the particle, that has smallest particle id in a tuple, became a key of tuple and is contained in `it->first`. The other particles are stored in `it->second`. Then, total particle number of a subchain equals as pList.size() + 1.